### PR TITLE
Fixes #12053: Add a "rudder agent history" command

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/12053

### CODE REVIEW

File: share/commands/agent-history [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# @man +
+# @man *-n*: show maximum n lines of history
+# @man +
+
+. "${BASEDIR}/../lib/common.sh"
+. "${BASEDIR}/../lib/cfengine_parsing.sh"
+
+BASE_PATH="/var/rudder/cfengine-community/outputs"
+# maximum 10000 lines
+max=25
```
Comment (ncharles) [inline share/commands/agent-history]: the link between the comment and the value is not obvious

Comment (peckpeck) [reply share/commands/agent-history]: indeed

**Comment by peckpeck** [comment]
Commit modified

File: share/commands/agent-history [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+done
+
+
+for file in $(ls -1 ${BASE_PATH}/cf* | sort | tail -n ${max})
+do
+  if type stat >/dev/null 2>/dev/null
+  then
+    endtime=$(stat -c "%Y" "${file}")
+  else
+    endtime=$(/opt/rudder/bin/perl -e'@a=stat $ARGV[0]; print $a[9]' "${file}")
```
Comment (amousset) [inline share/commands/agent-history]: thsi won't work on rudder-agent-thin?

Comment (peckpeck) [reply share/commands/agent-history]: platform supported by rudder-agent-thin have a working stat command

Comment (ncharles) [reply share/commands/agent-history]: yes, but there is no /opt/rudder/bin/perl on rudder-agent-thin systems

Comment (peckpeck) [reply share/commands/agent-history]: yes but there is a stat command on OS supported by rudder-agent thin, so we never reach this line

**Comment by peckpeck** [comment]
OK, merging this PR

**Pull request merged**